flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java

```diff
@@ -167,20 +163,25 @@
 	private JobGraph jobGraph;
 	private JobID jobId;
 
-	@Before
-	public void setUp() throws Exception {
-		MockitoAnnotations.initMocks(this);
+	private static final Configuration restConfig;
 
+	static {
 		final Configuration config = new Configuration();
 		config.setString(JobManagerOptions.ADDRESS, "localhost");
 		config.setInteger(RestOptions.RETRY_MAX_ATTEMPTS, 10);
 		config.setLong(RestOptions.RETRY_DELAY, 0);
+		config.setInteger(RestOptions.PORT, 0);
+
+		restConfig = config;
+	}
 
-		restServerEndpointConfiguration = RestServerEndpointConfiguration.fromConfiguration(config);
+	@Before
+	public void setUp() throws Exception {
+		restServerEndpointConfiguration = RestServerEndpointConfiguration.fromConfiguration(restConfig);
 		mockGatewayRetriever = () -> CompletableFuture.completedFuture(mockRestfulGateway);
 
 		executor = Executors.newSingleThreadExecutor(new ExecutorThreadFactory(RestClusterClientTest.class.getSimpleName()));
-		final RestClient restClient = new RestClient(RestClientConfiguration.fromConfiguration(config), executor) {
+		restClient = new RestClient(RestClientConfiguration.fromConfiguration(restConfig), executor) {
```

Comment:
I think we should only create a `RestClient` if we also create a `RestClusterClient`. Otherwise, we might not properly close the `RestClient` (e.g. in the `testRESTManualConfigurationOverride` test).